ouroboros-consensus/test-storage/Test/Ouroboros/Storage/VolatileDB/TestBlock.hs

```diff
@@ -35,35 +35,41 @@ import qualified Ouroboros.Storage.VolatileDB.Impl as Internal hiding (openDB)
 type BlockId = Word
 type SerialMagic = Int
 type Predecessor = WithOrigin BlockId
-type BlockTestInfo = (BlockId, Predecessor)
+type BlockTestInfo = (BlockId, Predecessor, BL.ByteString)
 
-type TestBlock = (BlockId, Predecessor, SerialMagic)
+type TestBlock = (BlockId, Predecessor, BL.ByteString, SerialMagic)
 
 toBinary :: BlockTestInfo -> BL.ByteString
 toBinary = Binary.encode . convert . toBlock
   where
-    convert :: TestBlock -> (BlockId, Maybe BlockId, SerialMagic)
-    convert (bid, pre, magic) = (bid, withOriginToMaybe pre, magic)
+    convert :: TestBlock -> (BlockId, Maybe BlockId, BL.ByteString, SerialMagic)
+    convert (bid, pre, h, magic) = (bid, withOriginToMaybe pre, h, magic)
 
 fromBinary :: BS.ByteString -> Either String BlockTestInfo
 fromBinary bs = do
     block <- decode bs
     case block of
-        (bid, predb, 1 :: SerialMagic) -> Right (bid, withOriginFromMaybe predb)
-        _                              -> Left "wrong payload"
+        (bid, predb, h, 1 :: SerialMagic) -> Right (bid, withOriginFromMaybe predb, h)
+        _                                 -> Left "wrong payload"
 
 guessSlot :: BlockId -> SlotNo
 guessSlot = SlotNo . fromIntegral
 
 toBlock :: BlockTestInfo -> TestBlock
-toBlock (b, pb) = (b, pb, 1)
+toBlock (b, pb, h) = (b, pb, h, 1)
 
 fromBlock :: TestBlock -> BlockTestInfo
-fromBlock (bid, pbid, 1) = (bid, pbid)
-fromBlock _              = error "wrong payload"
+fromBlock (bid, pbid, h, 1) = (bid, pbid, h)
+fromBlock _                 = error "wrong payload"
 
 binarySize :: Word64
-binarySize = 25
+binarySize = 34
+
+headerOffset :: Int
```

Comment:
Make this and `headerSize` `Word16`s, so you don't need `fromIntegral`s wherever you use them.